### DISCUSSION TITLE
Is it possible to add a syntactic sugar for generic constraints?

### DISCUSSION THREAD

**Discussion by fenginsc** [dotnet/csharplang#5084] [Language Ideas]
The existing syntax uses the where statement.
`
public struct Nullable<T> where T : struct { /*...*/ }
`
Can you provide a syntactic sugar to make the sentence more concise？
for example:
`
public struct Nullable<T : struct> { /*...*/ }
`

**Comment by CyrusNajmabadi** [comment]
it is highly unlikely that we would do this.  Brevity at the declaration point is just not that important given how rare and uncommon it is to have to write this out in full, and how little overhead the full form has.

**Reply by fenginsc** [reply]
Although it is not very common to add constraints, refer to kotlin when there is only one constraint without writing the where statement. I just want to say that this is also a place that can be improved.

**Reply by huoyaoyuan** [reply]
> how little overhead the full form has

Adding a new syntax is very expensive. A little improvement is usually not enough.
